public/app/features/provisioning/hooks/useGetResourceRepositoryView.ts

```diff
@@ -0,0 +1,57 @@
+import { skipToken } from '@reduxjs/toolkit/query/react';
+
+import { RepositoryView, useGetFrontendSettingsQuery } from 'app/api/clients/provisioning';
+
+import { useGetFolderQuery } from '../../../api/clients/folder';
+import { AnnoKeyManagerIdentity } from '../../apiserver/types';
+
+interface GetResourceRepositoryArgs {
+  name?: string; // the repository name
+  folderUid?: string; // folder we are targeting
+}
+
+// This is safe to call as a viewer (you do not need full access to the Repository configs)
+export const useGetResourceRepositoryView = ({
+  name,
+  folderUid,
+}: GetResourceRepositoryArgs): RepositoryView | undefined => {
+  const settings = useGetFrontendSettingsQuery();
+  const folderQuery = useGetFolderQuery(name || !folderUid ? skipToken : { name: folderUid });
+
+  if (!settings.data?.items) {
+    return undefined;
```

Comment:
🤔 potentially we should propagate the `isLoading`

Reply:
Yes, we need to expose the `isLoading` from here and use that in the new folder component, etc.